src/qt/bitcoingui.cpp

```diff
@@ -1125,6 +1131,25 @@ void BitcoinGUI::updateWalletStatus()
 }
 #endif // ENABLE_WALLET
 
+void BitcoinGUI::updateProxyIcon()
+{
+    std::string ip_port;
+    bool proxy_enabled = clientModel->getProxyInfo(ip_port);
+
+    if (proxy_enabled) {
+        QString ip_port_q = QString::fromStdString(ip_port);
+
+        if (labelProxyIcon->pixmap() == 0) {
```

Comment:
nit, IMO the label pixmap could be initialized when the label is created above.

Reply:
This is consistent with `labelWalletHDStatusIcon`, so I will prefer the approach taken